org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java

```diff
@@ -242,6 +245,12 @@ private void generateTypeCheck(BlockScope scope, CodeStream codeStream) {
 	}
 }
 
+private void setPatternIsTotalType() {
+	if (this.pattern != null) {
+		this.pattern.isTotalTypeNode = true;
```

Comment:
does `isTotalTypeNode` implement what 14.11.1.1 calls "A set of case elements, PCE, covers a type T"?

If so, then I don't think setting true is correct for any paths that use narrowing, IOW, when a conversion is not "unconditionally exact".